### PR TITLE
Added stable version of synapse dataplane APIs

### PR DESCRIPTION
**Pull request by xolve**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by JeffreyRichter — APPROVED** [review]

**Comment by ruowan** [comment]
@jianyexi Could you help to take a look at this PR? look autorest failed due to the model conflict.

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2020-12-01/roleAssignments.json [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+      "description": "Contains details when the response code indicates an error.",
+      "type": "object",
+      "properties": {
+        "error": {
+          "description": "The error details.",
+          "$ref": "#/definitions/ErrorResponse"
+        }
+      }
+    },
+    "ErrorResponse": {
```
Comment (jianyexi) [inline specification/synapse/data-plane/Microsoft.Synapse/stable/2020-12-01/roleAssignments.json:224]: This has conflict with "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse", please use the latter one .

**Comment by idear1203** [comment]
This is data plane API. @lmazuel , could you please help with a review?

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2020-12-01/roleAssignments.json (hunk 1 above)
Comment (xolve) [reply specification/synapse/data-plane/Microsoft.Synapse/stable/2020-12-01/roleAssignments.json]: Thanks @jianyexi for relative path to exact type. I get the error:

```UnhandledPromiseRejectionWarning: Error: '$.definitions.ErrorResponse.title' has incompatible values```

File: specification/synapse/data-plane/readme.md [hunk 2]
```diff
@@ -118,6 +118,37 @@ input-file:
...
+- Microsoft.Synapse/stable/2020-12-01/entityTypes/DataFlow.json
+- Microsoft.Synapse/stable/2020-12-01/entityTypes/Dataset.json
+- Microsoft.Synapse/stable/2020-12-01/entityTypes/LinkedService.json
+- Microsoft.Synapse/stable/2020-12-01/entityTypes/Notebook.json
+- Microsoft.Synapse/stable/2020-12-01/entityTypes/Pipeline.json
+- Microsoft.Synapse/stable/2020-12-01/entityTypes/SparkJobDefinition.json
+- Microsoft.Synapse/stable/2020-12-01/entityTypes/SqlScript.json
+- Microsoft.Synapse/stable/2020-12-01/entityTypes/Trigger.json
+```
+
```
Comment (idear1203) [inline specification/synapse/data-plane/readme.md:151]:
```suggestion
### Tag: package-monitoring-2020-12-01

These settings apply only when `--tag=package-monitoring-2020-12-01` is specified on the command line.

``` yaml $(tag) == 'package-artifacts-2020-12-01'
input-file:
- Microsoft.Synapse/stable/2020-12-01/artifacts.json
- Microsoft.Synapse/stable/2020-12-01/bigDataPools.json
- Microsoft.Synapse/stable/2020-12-01/checkAccessSynapseRbac.json
- Microsoft.Synapse/stable/2020-12-01/gitintegration.json
- Microsoft.Synapse/stable/2020-12-01/integrationRuntimes.json
- Microsoft.Synapse/stable/2020-12-01/library.json
- Microsoft.Synapse/stable/2020-12-01/managedPrivateEndpoints.json
- Microsoft.Synapse/stable/2020-12-01/monitoring.json
- Microsoft.Synapse/stable/2020-12-01/operations.json
- Microsoft.Synapse/stable/2020-12-01/roleAssignments.json
- Microsoft.Synapse/stable/2020-12-01/roleDefinitions.json
- Microsoft.Synapse/stable/2020-12-01/roles.json
- Microsoft.Synapse/stable/2020-12-01/sparkJob.json
- Microsoft.Synapse/stable/2020-12-01/sqlPools.json
- Microsoft.Synapse/stable/2020-12-01/workspace.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/DataFlow.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/Dataset.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/LinkedService.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/Notebook.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/Pipeline.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/SparkJobDefinition.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/SqlScript.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/Trigger.json
```
### Tag: package-artifacts-2020-12-01

These settings apply only when `--tag=package-artifacts-2020-12-01` is specified on the command line.

``` yaml $(tag) == 'package-artifacts-2020-12-01'
input-file:
- Microsoft.Synapse/stable/2020-12-01/artifacts.json
- Microsoft.Synapse/stable/2020-12-01/bigDataPools.json
- Microsoft.Synapse/stable/2020-12-01/gitintegration.json
- Microsoft.Synapse/stable/2020-12-01/integrationRuntimes.json
- Microsoft.Synapse/stable/2020-12-01/library.json
- Microsoft.Synapse/stable/2020-12-01/operations.json
- Microsoft.Synapse/stable/2020-12-01/sparkJob.json
- Microsoft.Synapse/stable/2020-12-01/sqlPools.json
- Microsoft.Synapse/stable/2020-12-01/workspace.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/DataFlow.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/Dataset.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/LinkedService.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/Notebook.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/Pipeline.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/SparkJobDefinition.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/SqlScript.json
- Microsoft.Synapse/stable/2020-12-01/entityTypes/Trigger.json
```

### Tag: package-access-control-2020-12-01

These settings apply only when `--tag=package-access-control-2020-12-01` is specified on the command line.

``` yaml $(tag) == 'package-access-control-2020-12-01'
- Microsoft.Synapse/stable/2020-12-01/checkAccessSynapseRbac.json
- Microsoft.Synapse/stable/2020-12-01/roleAssignments.json
- Microsoft.Synapse/stable/2020-12-01/roleDefinitions.json
- Microsoft.Synapse/stable/2020-12-01/roles.json
```

### Tag: package-vnet-2020-12-01

These settings apply only when `--tag=package-vnet-2020-12-01` is specified on the command line.

``` yaml $(tag) == 'package-vnet-2020-12-01'
- Microsoft.Synapse/stable/2020-12-01/managedPrivateEndpoints.json
```

### Tag: package-monitoring-2020-12-01

These settings apply only when `--tag=package-monitoring-2020-12-01` is specified on the command line.

``` yaml $(tag) == 'package-monitoring-2020-12-01'
- Microsoft.Synapse/stable/2020-12-01/monitoring.json
```
```

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2020-12-01/roles.json [hunk 3]
```diff
@@ -0,0 +1,170 @@
+{
```
Comment (idear1203) [inline specification/synapse/data-plane/Microsoft.Synapse/stable/2020-12-01/roles.json:1]: Is this file original from "2020-02-01-preview"? I don't think we need old SRBAC API any more.

**Review by idear1203 — APPROVED** [review]
Approved.

**Comment by idear1203** [comment]
I have two questions:
1. Has the new API version 2020-12-01 already been deployed worldwide?
2. Despite the API version change, is there any other change between these files and original ones?

**Comment by xolve** [comment]
> 1. Has the new API version 2020-12-01 already been deployed worldwide?
Yes.

> 2. Despite the API version change, is there any other change between these files and original ones?
No.

**Pull request merged**